### ISSUE TITLE
compatibility with Halcyon

### ISSUE THREAD

**Issue by fiatjaf** [hdevtools/hdevtools#7]
I'm trying to make hdevtools work with within a halcyon setup for three days now, and I have experienced the following problems:

  * hdevtools is installed with `halcyon install hdevtools`
  * Halcyon installs packages at `/app`, the "package-db" directory is at `/app/sandbox/i386-linux-ghc-7.8.4-packages.conf.d`
  * the first time I run `hdevtools check Main.hs` on my example app it outputs some errors, I think it has worked correctly, but it missed some other errors -- which I noticed when trying to compile
  * the Syntastic integration on vim has never worked, it just doesn't output anything, I tried messing up with Syntastic vars, but nothing makes it appear -- for additional information, `hlint` is installed at the same `halcyon install hlint` and it works on Syntastic. the command `:SyntasticCheck hdevtools` doesn't exist on vim, although it should (`:SyntasticCheck hlint`, for example, exists).
  * `vim-hdevtools` doesn't work either. I install it, but it inside vim it is as if I haven't installed. commands starting with `:Hdev...` don't exist.
  * I tried setting up `$GHC_PACKAGE_PATH` so I could run `ghci` manually and load my example app from it. On ghci, it works (only after setting `$GHC_PACKAGE_PATH`, before that ghci cannot find the imported packages and everything fails), but hdevtools it fails with this know error: https://github.com/hdevtools/hdevtools/issues/6
  * Now that I've removed `$GHC_PACKAGE_PATH`, when I try to run a normal `hdevtools check`, it fails to find ghc:

```shell
fiatjaf@spooner ~/c/littleapp> hdevtools --version
hdevtools: version 0.1.2.2 (ghc-7.8.4-i386-linux, cabal-1.18.1.5)
fiatjaf@spooner ~/c/littleapp> ghc --version
The Glorious Glasgow Haskell Compilation System, version 7.8.4
fiatjaf@spooner ~/c/littleapp> cabal --version
cabal-install version 1.20.0.3
using version 1.20.0.0 of the Cabal library 
fiatjaf@spooner ~/c/littleapp> hdevtools check Main.hs 
Cabal error: The program ghc version >=6.4 is required but it could not be found.
```

Please, help. I'm a super begginer on Haskell and I don't know what could be happening. Perhaps there's an issue with the package path and hdevtools is being ignored by Syntastic/vim becaus it is failing silently because of missing packages? If is this, maybe hdevtools could accept custom package paths somehow.

@mietek, do you have a solution to all this?

**Comment by mietek** [comment]
I’m sorry, I haven’t had the time to work on Halcyon for a while.  You probably want to use [Stack](http://haskellstack.org) instead.

**Comment by ch1bo** [comment]
Is using stack an option for you?

**Comment by fiatjaf** [comment]
Not currently. I have tried it, but I don't have enough memory nor disk space on my VPS for this now. Haskell never worked for me because of its heaviness and other cabal problems (the other times there wasn't any stack or halcyon available). It is very frustrating. I hope I'll get a better setup in some months, then I'll try again.

**Comment by ch1bo** [comment]
So am I understanding correctly that you have a haskell application running on a remote server and you now want to set up hdevtools as well on it?

**Comment by fiatjaf** [comment]
No, I'm just trying to learn Haskell by creating a small simple app and I thought having type and error information on the editor would help me a lot and actually let me graps the power of the language. I use this VPS for all development. Using my home computers is a bad idea, for me, for many reasons.

**Comment by ch1bo** [comment]
So where shall we go from here?

Besides that stack should not use more memory than simple compilation using cabal or ghc directly, hdevtools should find registered packages of the `ghc` and `ghc-pkg` of PATH. So to further debug these issues follow up with some more information, e.g.:
```
which ghc
ghc --version
which ghc-pkg
ghc-pkg list
cat <your cabal file>.cabal
ls <your project folder>
```

**Comment by ch1bo** [comment]
Any news on this? I just released hdevtools version `0.1.4.0` which improves stack support in finding the compiler and libs without `PATH` modifications.
